models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/MiniMaxChatClient.java

```diff
@@ -0,0 +1,364 @@
+			map.put("role", message.role().name());
+		}
+		if (choice.finishReason() != null) {
+			map.put("finishReason", choice.finishReason().name());
+		}
+		map.put("id", id);
+		return map;
+	}
+
+	@Override
+	public Flux<ChatResponse> stream(Prompt prompt) {
+
+		MiniMaxApi.ChatCompletionRequest request = createRequest(prompt, true);
+
+		return this.retryTemplate.execute(ctx -> {
+
+			Flux<MiniMaxApi.ChatCompletionChunk> completionChunks = this.miniMaxApi.chatCompletionStream(request);
+
+			// For chunked responses, only the first chunk contains the choice role.
+			// The rest of the chunks with same ID share the same role.
+			ConcurrentHashMap<String, String> roleMap = new ConcurrentHashMap<>();
+
+			// Convert the ChatCompletionChunk into a ChatCompletion to be able to reuse
+			// the function call handling logic.
+			return completionChunks.map(chunk -> chunkToChatCompletion(chunk)).map(chatCompletion -> {
+				try {
+					chatCompletion = handleFunctionCallOrReturn(request, ResponseEntity.of(Optional.of(chatCompletion)))
+						.getBody();
+
+					@SuppressWarnings("null")
+					String id = chatCompletion.id();
+
+					List<Generation> generations = chatCompletion.choices().stream().map(choice -> {
+						if (choice.message().role() != null) {
+							roleMap.putIfAbsent(id, choice.message().role().name());
+						}
+						String finish = (choice.finishReason() != null ? choice.finishReason().name() : "");
+						var generation = new Generation(choice.message().content(),
+								Map.of("id", id, "role", roleMap.get(id), "finishReason", finish));
+						if (choice.finishReason() != null) {
+							generation = generation.withGenerationMetadata(
+									ChatGenerationMetadata.from(choice.finishReason().name(), null));
+						}
+						return generation;
+					}).toList();
+
+					return new ChatResponse(generations);
+				}
+				catch (Exception e) {
+					logger.error("Error processing chat completion", e);
+					return new ChatResponse(List.of());
+				}
+
+			});
+		});
+	}
+
+	/**
+	 * Convert the ChatCompletionChunk into a ChatCompletion. The Usage is set to null.
+	 * @param chunk the ChatCompletionChunk to convert
+	 * @return the ChatCompletion
+	 */
+	private MiniMaxApi.ChatCompletion chunkToChatCompletion(MiniMaxApi.ChatCompletionChunk chunk) {
+		List<MiniMaxApi.ChatCompletion.Choice> choices = chunk.choices().stream().map(cc -> {
+			MiniMaxApi.ChatCompletionMessage delta = cc.delta();
+			if (delta == null) {
+				delta = new MiniMaxApi.ChatCompletionMessage("", MiniMaxApi.ChatCompletionMessage.Role.ASSISTANT);
+			}
+			return new MiniMaxApi.ChatCompletion.Choice(cc.finishReason(), cc.index(), delta, cc.logprobs());
+		}).toList();
+
+		return new MiniMaxApi.ChatCompletion(chunk.id(), choices, chunk.created(), chunk.model(),
+				chunk.systemFingerprint(), "chat.completion", null, null);
+	}
+
+	/**
+	 * Accessible for testing.
+	 */
+	MiniMaxApi.ChatCompletionRequest createRequest(Prompt prompt, boolean stream) {
```

Comment:
In this method, if the model supports media, it needs to be added.  I guess the model does support media which is why the `fromMediaData` method is implemented.  Perhaps it just needs to be called here?

Reply:
MiniMax does not support media. The `fromMediaData` method designed for future model support has been removed in PR #744 .